File: backend/app/schemas/ontology.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class OntologyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    industry: str
    concept: str
    synonyms: list | None
    tech_stack: list | None
    prerequisites: list | None
    key_positions: list | None
    skill_relations: dict | None
    jargon: dict | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/ontology.ts
export interface OntologyResponse {
  id: string
  industry: string
  concept: string
  synonyms: string[] | null
  tech_stack: string[] | null
  prerequisites: string[] | null
  key_positions: string[] | null
  skill_relations: Record<string, string[]> | null
  jargon: Record<string, string> | null
  created_at: string
  updated_at: string
}
